src/library/scala/collection/mutable/HashSet.scala

```diff
@@ -104,7 +104,9 @@ extends AbstractSet[A]
  *  @define coll mutable hash set
  */
 object HashSet extends MutableSetFactory[HashSet] {
-  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, HashSet[A]] = setCanBuildFrom[A]
+  implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, HashSet[A]] =
+    ReusableCBF.asInstanceOf[CanBuildFrom[Coll, A, HashSet[A]]]
+  private[this] val ReusableCBF = setCanBuildFrom[Any]
```

Comment:
How about we make `GenSetFactory.setCanBuildFrom` backed by a val, and leave this as is? There might be a few other places where we the same approach works.

Reply:
doesnt it need to be able to refer to newBuilder etc in this scope?

Reply:
Yes, I misunderstood the code.